### PR TITLE
Fix heartbeat v2 metrics update

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- connections metrics are not updated on heartbeat v2 monitor
- node type metrics are not updated on heartbeat v2 sender
  
## Proposed Changes
- add metrics update

## Testing procedure
- system test
- scenario: after epoch 3, check metrics on `erd_live_validator_nodes` and `erd_connected_nodes`, `erd_node_type`, `erd_peer_type`, `erd_peer_subtype`. They should display the actual values, not 0.

### CODE REVIEW

File: heartbeat/monitor/monitor.go [hunk 1]
```diff
@@ -213,6 +223,29 @@ func (monitor *heartbeatV2Monitor) shouldSkipMessage(messageAge time.Duration, p
 	return false
 }
 
+func (monitor *heartbeatV2Monitor) updateMetrics(_ []byte, _ interface{}) {
+	heartbeats := monitor.GetHeartbeats()
+
+	counterActiveValidators := 0
+	counterConnectedNodes := 0
+	for _, heartbeatMessage := range heartbeats {
```
Comment (iulianpascalau) [inline heartbeat/monitor/monitor.go:235]: this is so sub-optimal, do not have a better idea, this is how the v1 was implemented.

Comment (iulianpascalau) [reply heartbeat/monitor/monitor.go:235]: Wondering if we can have a dedicated go routine that will call this each 30 seconds or so.

Comment (sstanculeanu) [reply heartbeat/monitor/monitor.go:263]: indeed this would be lighter..pushed

File: factory/heartbeatV2Components.go [hunk 2]
```diff
@@ -230,6 +233,10 @@ func (hc *heartbeatV2Components) Close() error {
 		log.LogIfError(hc.directConnectionsProcessor.Close())
 	}
 
+	if !check.IfNil(hc.monitor) {
+		log.LogIfError(hc.monitor.Close())
```
Comment (iulianpascalau) [inline factory/heartbeatV2Components.go:237]: :+1:

File: testscommon/cacherMock.go [hunk 3]
```diff
@@ -102,6 +102,9 @@ func (cacher *CacherMock) Remove(key []byte) {
 
 // Keys -
 func (cacher *CacherMock) Keys() [][]byte {
+	cacher.mut.RLock()
```
Comment (iulianpascalau) [inline testscommon/cacherMock.go:105]: :+1:

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**